File: javascript/hamming/hamming.js
//
// This is only a SKELETON file for the 'Hamming' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const compute = (strand1, strand2) => {
  if (strand1.length !== strand2.length) {
    throw new Error('strands must be of equal length');
  };
  return strand1.split('').filter((letter, index) => strand2[index] !== letter).length;
};

File: cpp/hamming/hamming.cpp
#include "hamming.h"
#include <string>
#include <stdexcept>

namespace hamming {
    int compute(std::string strand_1, std::string strand_2) {
        if (strand_1.size() != strand_2.size()) {
            throw std::domain_error("strands must have same length");
        }
        int count{0};
        for (int i = 0; i < int(strand_1.size()); i++) {
            if (strand_1[i] != strand_2[i]) count++;
        }
        return count;
    }
}  // namespace hamming
